mat_dp_pipeline/pipeline/pipeline.py

```diff
@@ -119,28 +121,40 @@ def __len__(self) -> int:
         return self._length
 
 
+def make_iterator(flattened: list[tuple[Path, SparseYearsInput]]):
+    for path, sparse_years in flattened:
+        for path, year, inpt in to_processable_input(path, sparse_years):
+            yield (path, year, inpt)
+
+
+def full_calculate(full_inpt: tuple[Path, Year, ProcessableInput]):
```

Comment:
Name is a bit awkward. Maybe it would be better if defined within `pipeline` something like `calculate_helper`? Don't know. Alternatively it could do with an underscore. Also return type would be nice.